src/components/utils/popover/popover-abstract.ts

```diff
@@ -259,16 +212,36 @@ export abstract class PopoverAbstract<Nodes extends PopoverNodes = PopoverNodes>
 
     if (item.closeOnActivate) {
       this.hide();
+
+      this.emit(PopoverEvent.CloseOnActivate, item);
+    }
+  }
+
+  /**
+   * Handles clicks inside popover
+   *
+   * @param event - item to handle click of
+   */
+  private handleClick(event: Event): void {
+    const item = this.getTargetItem(event);
+
+    if (item === undefined) {
+      return;
     }
+
+    this.handleItemClick(item);
   }
 
   /**
-   * Toggles nothing found message visibility
+   * Calls .preventDefault() on 'mousedown' event to prevent unwanted focus loss in safari
    *
-   * @param isDisplayed - true if the message should be displayed
+   * @param event - mousedown event
    */
-  private toggleNothingFoundMessage(isDisplayed: boolean): void {
-    this.nodes.nothingFoundMessage.classList.toggle(css.nothingFoundMessageDisplayed, isDisplayed);
+  private handleMousedown(event: Event): void {
+    /**
+     * preventDefault() is needed to prevent unwanted focus loss in safari
```

Comment:
Describe (in jsdoc) why the focus is losing please

Reply:
outdated